parsers/verilog.c

```diff
@@ -873,12 +873,30 @@ static int _readWordToken (tokenInfo *const token, int c, bool skip)
 	Assert (isWordToken (c));
 
 	clearToken (token);
-	do
+	if (c == '\\')
 	{
-		vStringPut (token->name, c);
-		c = vGetc ();
-	} while (isIdentifierCharacter (c));
-	_updateKind (token);
+		// Escaped identifier
+		c = vGetc (); // skip leading '\'
+		// A single `\` would result in an empty identifier, which is unsupported.
+		// Add the initial `\` in that case, and also in case it starts with `\\`.
+		if (!isgraph (c) || c == '\\')
```

Comment:
I wonder if C is EOF (== -1).
isgraph(c) may return false. Then `\` will be appended to `token->name`. It's not so harmful that I am afraid of. But putting something like `goto end;` will be better.

Reply:
I think the behavior is correct.

We treat the sequence `\<EOF>` the same as `\<whitespace>`, i.e. the token finishes right there so it's just a single `"\"`.  This would generate an empty identifier which ctags doesn't like, so as suggested by @roccomao we add the leading `\` to that corner case, and also to all identifier names starting with `\`.  Note that this is only done for the first character.

If I'm interpreting [the C standard, §7.4](https://www.iso-9899.info/n1570.html#7.4) correctly, passing EOF to `isgraph()` et al is valid and would return false, since EOF is not a graph character.  So EOF will be treated the same as space, whitespace, control characters, etc., which is what we want.

(PS: Note that I'm using `while` and not `do while` to handle this case, since the sequence of characters after the `\` may be empty, whereas in the case of regular identifiers there's one or more characters, hence `do while`)

Reply:
I see.